pkg/agent/openflow/pipeline.go

```diff
@@ -1622,8 +1622,13 @@ func (c *client) serviceLearnFlow(groupID binding.GroupIDType, svcIP net.IP, svc
 		MatchProtocol(protocol).
 		MatchDstPort(svcPort, nil).
 		Cookie(cookieID)
+	// affinityTimeout is used as the OpenFlow "hard timeout": learned flow will be removed from
+	// OVS after that time regarding of whether traffic is still hitting the flow. This is the
+	// desired behavior based on the K8s spec. Note that existing connections will keep going to
+	// the same endpoint because that's how OVS group selection works; and that is also the
```

Comment:
Is it because how OVS group selection works or the pipeline executes `ct(nat)` in table 30? I'm not sure what OVS group selection do if we don't do nat in table 30, but from dp flows the subsequent packets are not processed by group.
The first packet:
```
recirc_id(0),in_port(6),eth(src=ca:63:74:cd:ca:5d),eth_type(0x0800),ipv4(src=172.60.0.68,dst=0.0.0.0/128.0.0.0,frag=no), packets:7, bytes:1318, used:2.592s, flags:PR., actions:ct(zone=65520,nat),recirc(0x29c)
recirc_id(0x29c),in_port(6),ct_state(+new-inv+trk),ct_mark(0),eth(),eth_type(0x0800),ipv4(dst=10.96.0.1,proto=6,frag=no),tcp(dst=443), packets:0, bytes:0, used:never, actions:hash(l4(0)),recirc(0x29d)
recirc_id(0x29d),dp_hash(0xc/0xf),in_port(6),eth(),eth_type(0x0800),ipv4(proto=6,frag=no), packets:0, bytes:0, used:never, actions:ct(commit,zone=65520,mark=0x21/0xffffffff,nat(dst=192.168.10.4:6443)),recirc(0x29e)
recirc_id(0x29e),in_port(6),ct_state(+new-est+trk),ct_mark(0x21),eth(dst=46:c9:e8:7e:7f:24),eth_type(0x0800),ipv4(src=172.60.0.64/255.255.255.192,dst=192.0.0.0/192.0.0.0,frag=no), packets:0, bytes:0, used:never, actions:2
```
The subsequent packets:
```
recirc_id(0),in_port(6),eth(src=ca:63:74:cd:ca:5d),eth_type(0x0800),ipv4(src=172.60.0.68,dst=0.0.0.0/128.0.0.0,frag=no), packets:7, bytes:1318, used:2.592s, flags:PR., actions:ct(zone=65520,nat),recirc(0x29c)
recirc_id(0x29c),in_port(6),ct_state(-new+est+trk),ct_mark(0x21),eth(dst=46:c9:e8:7e:7f:24),eth_type(0x0800),ipv4(dst=192.0.0.0/192.0.0.0,frag=no), packets:6, bytes:1252, used:2.592s, flags:PR., actions:2
```

Reply:
You're right, I fixed the comment. My original statement doesn't apply when using connection tracking / nat